src/main/java/hudson/plugins/copyartifact/CopyArtifact.java

```diff
@@ -497,7 +497,7 @@ private boolean perform(Run src, Run<?,?> dst, String expandedFilter, String exp
     public static final class DescriptorImpl extends BuildStepDescriptor<Builder> {
 
         public FormValidation doCheckProjectName(
-                @AncestorInPath AbstractProject anc, @QueryParameter String value) {
+                @AncestorInPath Job<?,?> anc, @QueryParameter String value) {
```

Comment:
Right, this should help with the excepton.

The bogus object is something else, typically caused by some nonstandard databinding idiom, missing getter, etc.; but possibly a bug in Workflow in terms of handling unusual collection signatures or the like. You can add a `test`-scoped dependency on `workflow-api` and call `DescribableHelper.uninstantiate` on a configured `CopyArtifact` to see what went wrong. (Look at `DescribableHelperTest` for examples of the quasi-JSON expected results. There should be no Java objects other than primitive wrappers, `String`, `List`, and `Map`.)